tests/kernel/io_uring.pm

```diff
@@ -86,13 +88,23 @@ sub run {
     );
 
     # search for timed out tests
-    @lines = $out =~ /Tests timed out \(d+\):.*/mg;
-    if (@lines) {
-        record_info(
-            "Timeout",
-            $lines[0],
-            result => 'softfail'
-        );
+    my @timeouts;
+    for my $line ($out =~ /Tests timed out \(\d+\):.*/mg) {
+        push @timeouts, $line =~ /<([\w\-\.]+\.t)>/g;
+    }
+    if (@timeouts) {
+        record_info("Timed-out Tests", join(", ", @timeouts));
+        for my $timeout (@timeouts) {
+            unless ($whitelist->override_known_failures(
```

Comment:
You could also do `next if $whitelist->override_known_failures(...);` instead of wrapping the `record_info()` below in another conditional block. But LGTM as is.

Reply:
Though maybe rename `$timeout` to `$testname` so it's more legible.